src/algoliasearch.helper.js

```diff
@@ -1215,6 +1215,29 @@ AlgoliaSearchHelper.prototype.clearCache = function() {
 };
 
 /**
+ * Updates the internal client instance. If the reference of the clients
+ * are equal then no update is actually done.
+ * @param  {AlgoliaSearch} newClient an AlgoliaSearch client
+ * @return {AlgoliaSearchHelper}
+ */
+AlgoliaSearchHelper.prototype.setClient = function(newClient) {
+  if (this.client === newClient) return this;
+
+  if (!newClient.addAlgoliaAgent) newClient.addAlgoliaAgent('JS Helper ' + version);
```

Comment:
Should you call setClient from the constructor now that you have this init method?